tests/core/test_path_guard.py

```
@@ -0,0 +1,388 @@
+
+    def test_resolve_runtime_error_is_wrapped_as_path_traversal(
+        self, tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+    ) -> None:
+        """Cyclic symlinks make `Path.resolve()` raise `RuntimeError`, not
+        `ValueError`. Callers expect `PathTraversalError` / `ValueError`
+        uniformly, so the helper must wrap resolver failures.
+        """
+        path = tmp_path / "cyclic"
+        original_resolve = Path.resolve
+
+        def raise_cycle(self: Path, *args, **kwargs):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `raise_cycle`

(ANN202)